### PR TITLE
Use Preact in Production

### PR DESCRIPTION
**Pull request by chrishelgert**
* get the benefits from the react ecosystem (hot loading, profiling) in development
* smaller bundle in production
* reducing complete page size from 525 kb to 396kb

> see official example https://github.com/zeit/next.js/tree/master/examples/using-preact

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -27,19 +27,23 @@
...
     "babel-plugin-styled-components": "^1.1.7",
+    "cross-env": "^5.0.5",
     "dotenv-webpack": "^1.5.4",
     "isomorphic-unfetch": "^2.0.0",
     "js-base64": "^2.1.9",
+    "module-alias": "^2.0.1",
     "next": "^3.0.6",
     "polished": "^1.6.1",
-    "react": "^15.6.1",
-    "react-dom": "^15.6.1",
```
Comment (danielbayerlein) [inline package.json]: Is `react` and `react-dom` not required? See https://github.com/zeit/next.js/pull/2583

Comment (chrishelgert) [reply package.json]: in production only preact and preact-compat is required, there for react and react-dom will only be shipped via yarn/npm in dev :)

if you clean your node_modules and  reinstall it with yarn install --prod, then it also works with yarn build and yarn start :)

Comment (danielbayerlein) [reply package.json]: Merge error or do we need both dependencies?

```
"preact": "^8.2.5",
"preact-compat": "^3.17.0",
"react": "^15.6.1",
"react-dom": "^15.6.1"
```

Comment (chrishelgert) [reply package.json]: merge error - i fixed it :)

**Comment by danielbayerlein** [comment]
Next.js + preact: https://github.com/zeit/next-plugins/tree/master/packages/next-preact

**Pull request closed without merging**